static/examples/audio.html

```diff
@@ -0,0 +1,188 @@
+            return new Promise(function(resolve, reject) {
+              getUserMedia.call(navigator, constraints, resolve, reject);
+            });
+          }
+        }
+
+        let audioCtx = new (window.AudioContext || window.webkitAudioContext)();
+        let voiceSelect = document.getElementById("voice");
+        let source, stream;
+
+        /* ______________       _______________
+        * | Event Target | <--- | AudioContext |
+        * |  ____________|      |______________|
+         */
+
+        let distortion = audioCtx.createWaveShaper();
+        let gainNode = audioCtx.createGain();
+        let biquadFilter = audioCtx.createBiquadFilter();
+        let delay = audioCtx.createDelay();
+
+        // distortion curve for the waveshaper, thanks to Kevin Ennis
+        // http://stackoverflow.com/questions/22312841/waveshaper-node-in-webaudio-how-to-emulate-distortion
+        // 
+        // it would be cool to change this into a vocoder / DSP for autotuning
+        // http://blogs.zynaptiq.com/bernsee/pitch-shifting-using-the-ft/
+        // see also https://github.com/k15z/pitch-perfect-auto-tune
+
+        function makeDistortionCurve(amount) {
+          let k = typeof amount === 'number' ? amount : 50,
+            n_samples = 44100,
+            curve = new Float32Array(n_samples),
+            deg = Math.PI / 180,
+            i = 0,
+            x;
+          for ( ; i < n_samples; ++i ) {
+            x = i * 2 / n_samples - 1;
+            curve[i] = ( 3 + k ) * x * 20 * deg / ( Math.PI + k * Math.abs(x) );
+          }
+          return curve;
+        };
+
+
+        if (navigator.mediaDevices.getUserMedia) {
+           const constraints = {
+            audio: {
+              echoCancellation: true,
+              noiseSupression: false,
+              volume: {
+                exact: {
+                  min: 0,
+                  max: 90,
+                },
+                ideal: {
+                  min: 0,
+                  max: 90,
+                }
+              }
+            }
+          }
+          navigator.mediaDevices.getUserMedia(constraints)
+            .then(function(stream) {
+              source = audioCtx.createMediaStreamSource(stream);
+              source.connect(distortion);
+              distortion.connect(biquadFilter);
+              biquadFilter.connect(gainNode);
+              gainNode.connect(delay);
+              delay.connect(audioCtx.destination);
+
+              range.oninput = function() {
+                gainNode.gain.value = range.value;
+              }
+              // Delay time needs to between 0, 1 but the slider only returns integers
+              delayrange.oninput = function() {
+                delay.delayTime.value = delayrange.value / 10;
+              }
+              voiceChange();
+            })
+            .catch( function(err) { console.log('The following gUM error occured: ' + err);})
```

Comment:
On Safari, I get the following error when I `Start Microphone`:

> The following gUM error occured: TypeError: The provided value is non-finite - audio.html:114

- Safari 13.2 and Safari 13.1 both fail this way
- Works fine in Chrome/FF